command/agent/command.go

```diff
@@ -76,6 +76,10 @@ func (c *Command) readConfig() *Config {
 	cmdFlags.Var((*AppendSliceValue)(&tags), "tag",
 		"tag pair, specified as key=value")
 	cmdFlags.StringVar(&cmdConfig.Discover, "discover", "", "mDNS discovery name")
+	cmdFlags.Var((*AppendSliceValue)(&cmdConfig.SRVRecords), "srvrecord",
```

Comment:
I understand the comment about join-srv VS retry-join-srv, but I don't see how I can reuse anything to do with the wan / lan profiles, because they don't seem to have anything to do with joining:

https://github.com/hashicorp/memberlist/blob/9631bcf2ffb1632b05d00d7a756707dd15d81b86/config.go#L188-L198

The profiles only seem to be concerned with the actual health checking of nodes. Could you demonstrate where join changes its behavior based on wan/lan? I don't really get it.